Fix for Oracle bug 16027951 Xorg GUI fails to start when installing solaris 11.

Root cause: ast driver uses only 128 bytes buffer for EDID reading.
When extension flag (the 126th byte) is non-zero, indicating
extensions blocks to follow, X server reads beyond the buffer
boundary resulting in seg fault.

As discussed in the Oracle bug report, throwing away the extension data isn't
great, but a more complete fix requires reworking the code to dynamically
allocate more than 128 bytes when the extension data is present, so this patch
isn't suitable for upstream in it's current form.

Reported upstream:
https://gitlab.freedesktop.org/xorg/driver/xf86-video-ast/-/issues/6


--- xf86-video-ast-1.2.0/src/ast_driver.c
+++ xf86-video-ast-1.2.0/src/ast_driver.c
@@ -1504,6 +1504,12 @@ }
       else
           Flags = ASTGetVGAEDID(pScrn, DDC_data);
 
+      /*
+       * Buffer is of size 128, can't read extension blocks even the monitor
+       * provides. 126 is NO_EDID defined in edid.h
+       */
+      DDC_data[126] = 0;
+
       if (Flags)
       {
           ConfiguredMonitor = xf86InterpretEDID(pScrn->scrnIndex, DDC_data);
@@ -1556,6 +1562,12 @@ }
       else
           Flags = ASTGetVGAEDID(pScrn, DDC_data);
 
+      /*
+       * Buffer is of size 128, can't read extension blocks even the monitor
+       * provides. 126 is NO_EDID defined in edid.h
+       */
+      DDC_data[126] = 0;
+
       if (Flags)
       {
           MonInfo = MonInfo1 = xf86InterpretEDID(pScrn->scrnIndex, DDC_data);
